SignalServiceKit/src/Storage/OWSDatabaseConnection.m

```diff
@@ -47,32 +47,13 @@ - (void)readWithBlock:(void (^)(YapDatabaseReadTransaction *transaction))block
 
 - (void)asyncReadWithBlock:(void (^)(YapDatabaseReadTransaction *transaction))block
 {
-    id<OWSDatabaseConnectionDelegate> delegate = self.delegate;
-    OWSAssert(delegate);
-    OWSAssert(delegate.areSyncRegistrationsComplete);
-
-    [delegate readTransactionWillBegin];
-    [super asyncReadWithBlock:block
-              completionBlock:^{
-                  [delegate readTransactionDidComplete];
-              }];
+    [self asyncReadWithBlock:block completionQueue:NULL completionBlock:nil];
```

Comment:
The base class implementation just defers to the all-params flavor of this method. Do that explicitly here to DRY up the logic in this class.